pkg/knit/cmd/k8s/podinfo.go

```diff
@@ -140,17 +139,17 @@ func getKubeConfig() (*rest.Config, error) {
 		return kubeconfigFromFilePath(kubeConfig)
 	}
 
+	// try the in-cluster config
```

Comment:
This one 
- incoming parameter ( if exists) 
- KUBECONFIG env var
- in-cluster config
- home/.kube/config

That is why I have changed it, because if KUBECONFIG was not defined it try to check for "home/.kube/config" and fails without checking for the in-cluster config.

Reply:
ok, and which tooling/client did we check?

Reply:
Maybe a better option could be to use ["sigs.k8s.io/controller-runtime/pkg/client/config" GetConfig() funcion](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/client/config#GetConfig)

Reply:
> ok, and which tooling/client did we check?

Sorry did  not see it ... I took a look at GetConfig in the controller-runtime and to some other github repos doing the same

Reply:
following the best practices is obviously fine to me. We just need to document where we learned the best practice, and this is good enough indeed.
The real issue seems to lie here:
```
if usr, err := user.Current(); err == nil {
		kubeConfig := filepath.Join(usr.HomeDir, ".kube", "config")
		return kubeconfigFromFilePath(kubeConfig)
}
```
we should rather follow through if the read from kubeconfig file fails - and not abort the function; that would have allowed the code to try the `InClusterConfig`, perhaps annoyingly (?) emitting a warning in the logs.

However, following the same order like controller-runtime does seems a better approach overall, so let's do it.

Reply:
/hold